plugins/drivers/base/utils.go

```diff
@@ -0,0 +1,298 @@
+		StartedAt:    started,
+		CompletedAt:  completed,
+		ExitResult:   exitResultFromProto(pb.Result),
+	}, nil
+}
+
+func taskStatsToProto(stats *TaskStats) (*proto.TaskStats, error) {
+	timestamp, err := ptypes.TimestampProto(time.Unix(stats.Timestamp, 0))
+	if err != nil {
+		return nil, err
+	}
+
+	pids := map[string]*proto.TaskResourceUsage{}
+	for pid, ru := range stats.ResourceUsageByPid {
+		pids[pid] = resourceUsageToProto(ru)
+	}
+
+	return &proto.TaskStats{
+		Id:                 stats.ID,
+		Timestamp:          timestamp,
+		AggResourceUsage:   resourceUsageToProto(stats.AggResourceUsage),
+		ResourceUsageByPid: pids,
+	}, nil
+}
+
+func taskStatsFromProto(pb *proto.TaskStats) (*TaskStats, error) {
+	timestamp, err := ptypes.Timestamp(pb.Timestamp)
+	if err != nil {
+		return nil, err
+	}
+
+	pids := map[string]*cstructs.ResourceUsage{}
+	for pid, ru := range pb.ResourceUsageByPid {
+		pids[pid] = resourceUsageFromProto(ru)
+	}
+
+	stats := &TaskStats{
+		ID:                 pb.Id,
+		Timestamp:          timestamp.Unix(),
+		AggResourceUsage:   resourceUsageFromProto(pb.AggResourceUsage),
+		ResourceUsageByPid: pids,
+	}
+
+	return stats, nil
+}
+
+func resourceUsageToProto(ru *cstructs.ResourceUsage) *proto.TaskResourceUsage {
+	cpu := &proto.CPUUsage{}
+	for _, field := range ru.CpuStats.Measured {
+		switch field {
+		case "System Mode":
+			cpu.SystemMode = ru.CpuStats.SystemMode
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_SYSTEM_MODE)
+		case "User Mode":
+			cpu.UserMode = ru.CpuStats.UserMode
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_USER_MODE)
+		case "Total Ticks":
+			cpu.TotalTicks = ru.CpuStats.TotalTicks
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_TOTAL_TICKS)
+		case "Throttled Periods":
+			cpu.ThrottledPeriods = ru.CpuStats.ThrottledPeriods
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_THROTTLED_PERIODS)
+		case "Throttled Time":
+			cpu.ThrottledTime = ru.CpuStats.ThrottledTime
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_THROTTLED_TIME)
+		case "Percent":
+			cpu.Percent = ru.CpuStats.Percent
+			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_PERCENT)
+		}
+	}
+
+	memory := &proto.MemoryUsage{}
+	for _, field := range ru.MemoryStats.Measured {
+		switch field {
+		case "RSS":
+			memory.Rss = ru.MemoryStats.RSS
+			memory.MeasuredFields = append(memory.MeasuredFields, proto.MemoryUsage_RSS)
+		case "Cache":
+			memory.Cache = ru.MemoryStats.Cache
+			memory.MeasuredFields = append(memory.MeasuredFields, proto.MemoryUsage_CACHE)
+		case "Max Usage":
+			memory.MaxUsage = ru.MemoryStats.MaxUsage
+			memory.MeasuredFields = append(memory.MeasuredFields, proto.MemoryUsage_MAX_USAGE)
+		case "Kernel Usage":
+			memory.KernelUsage = ru.MemoryStats.KernelUsage
+			memory.MeasuredFields = append(memory.MeasuredFields, proto.MemoryUsage_KERNEL_USAGE)
+		case "Kernel Max Usage":
```

Comment:
We could really use consts for these... not a problem for this PR though :)

Reply:
Yeah I agree we don't have them anywhere currently and I wasn't sure if this package was the best place.